dpnp/dpnp_iface_mathematical.py

```diff
@@ -387,6 +388,71 @@ def ceil(
     )
 
 
+def clip(a, a_min, a_max, *, out=None, order="K"):
+    """
+    Clip (limit) the values in an array.
+
+    For full documentation refer to :obj:`numpy.clip`.
+
+    Parameters
+    ----------
+    a : {dpnp_array, usm_ndarray}
+        Array containing elements to clip.
+    a_min, a_max : {dpnp_array, usm_ndarray, None}
+        Minimum and maximum value. If ``None``, clipping is not performed on the corresponding edge
+        Only one of `a_min` and `a_max` may be ``None``. Both are broadcast against a.
+    out : dpnp_array, optional
+        The results will be placed in this array. It may be the input array for in-place clipping.
+        `out` must be of the right shape to hold the output. Its type is preserved.
+
+    Returns
+    -------
+    out : dpnp_array
+        An array with the elements of `a`, but where values < `a_min` are replaced with `a_min`,
+        and those > `a_max` with `a_max`.
+
+    Examples
+    --------
+    >>> import dpnp as np
+    >>> a = np.arange(10)
+    >>> a
+    array([0, 1, 2, 3, 4, 5, 6, 7, 8, 9])
+    >>> np.clip(a, 1, 8)
+    array([1, 1, 2, 3, 4, 5, 6, 7, 8, 8])
+    >>> np.clip(a, 8, 1)
+    array([1, 1, 1, 1, 1, 1, 1, 1, 1, 1])
+    >>> np.clip(a, 3, 6, out=a)
+    array([3, 3, 3, 3, 4, 5, 6, 6, 6, 6])
+    >>> a
+    array([3, 3, 3, 3, 4, 5, 6, 6, 6, 6])
+
+    >>> a = np.arange(10)
+    >>> a
+    array([0, 1, 2, 3, 4, 5, 6, 7, 8, 9])
+    >>> min = np.array([3, 4, 1, 1, 1, 4, 4, 4, 4, 4])
+    >>> np.clip(a, min, 8)
+    array([3, 4, 2, 3, 4, 5, 6, 7, 8, 8])
+
+    """
+
+    usm_arr = dpnp.get_usm_ndarray(a)
+    usm_min = (
+        dpnp.get_usm_ndarray(a_min) if isinstance(a_min, dpnp_array) else a_min
+    )
+    usm_max = (
+        dpnp.get_usm_ndarray(a_max) if isinstance(a_max, dpnp_array) else a_max
+    )
+
+    if out is not None:
+        usm_out = dpnp.get_usm_ndarray(out)
+        dpt.clip(usm_arr, usm_min, usm_max, out=usm_out, order=order)
```

Comment:
dpctl doesn't support `order=None`, but numpy does and we need to handle that also